Document package management limitations (#885)
digital-asset/docs.daml.com: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

-  Contract keys in upgradable packages can only include types defined
   within the same package, or types from the Daml Standard Library.

> There are further limitations with respect to managing the packages on a running ledger

Reviewer: ```suggestion
There are further limitations with respect to managing the packages on a running ledger:
```

- Once a version of a package is uploaded to the ledger, it cannot be replaced or removed.
  If a package is uploaded by mistake it can only be overriden by uploading an even newer version.

- As a consequence of the above, if a record is extended by mistake with an optional field,

>   that field will need to be part of all future versions.

Reviewer: ```suggestion
  that field must be part of all future versions.
```
